### PR TITLE
added a tutorial for gathering metrics

### PR DESCRIPTION
**Pull request by jdufner**

### CODE REVIEW

**Comment by obfischer** [comment]
Hi @jdufner, I will have a look on it this evening.

File: src/main/tutorials/calculate-metrics/index.adoc [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+  If you can't measure it, you can't change it.
+
+  You can't manage what you can't measure.
+
+  If it cannot be measured, it cannot be managed.
+
+////
+
+Metrics are just one way to measure you software system.
```
Comment (obfischer) [inline src/main/tutorials/calculate-metrics/index.adoc]: Did you mean _measure your software system_?

File: src/main/tutorials/calculate-metrics/index.adoc [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+They can help you to understand your system and to discover anomalies in your system.
+If the anomalies discovered, it's up to you to improve your system to fulfill the metrics.
+
+In the following chapters I describe different metrics and their purpose.
```
Comment (obfischer) [inline src/main/tutorials/calculate-metrics/index.adoc:43]: Is it your goal to describe the metrics and their purpose or would you like to show the reader how specific metrics can be calculated with jQAssistant? What would you like to focus on?

File: src/main/tutorials/calculate-metrics/jqassistant/ood-metrics.adoc [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+SET p.abstractness = toFloat(p.na) / p.nc
+RETURN p
+----
+
+==== Example
+
+.OO-Metrics Abstractness
+image::OO-Metrics_Abstractness.png[OO-Metrics Abstractness]
+
+==== Test cases
```
Comment (obfischer) [inline src/main/tutorials/calculate-metrics/jqassistant/ood-metrics.adoc]: The test cases are confusing for me? What should be tested here? Wouldn't it be better for the reader see how he can define constraints to ensure that specific thresholds won't be violated?

File: src/main/tutorials/calculate-metrics/pom.xml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>your.project</groupId>
+  <artifactId>metrics</artifactId>
+  <version>0.0.1-SNAPSHOT</version>
+  <name>metrics</name>
+  <description>Test project to evaluate the metrics.</description>
+
+  <dependencies>
```
Comment (obfischer) [inline src/main/tutorials/calculate-metrics/pom.xml]: Is this dependency needed?

File: src/main/tutorials/calculate-metrics/jqassistant/ood-metrics.adoc [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+.Normalized distance Package c
+[source,cypher,role=concept,requiresConcepts="ood-metrics:NormalizedDistance",severity=major]
+----
+MATCH (p:Package {fqn:"your.project.oodmetrics.c", normalizedDistance:0.25})
+RETURN p
+----
+
+[[ood-metrics:NormalizedDistancePackageD]]
+.Normalized distance Package D
+[source,cypher,role=concept,requiresConcepts="ood-metrics:NormalizedDistance",severity=major]
```
Comment (obfischer) [inline src/main/tutorials/calculate-metrics/jqassistant/ood-metrics.adoc]: Attributes like `role` or `requiresConcepts` not visible in the rendered tutorial. Users not so familiar with jQAssistant will not be able to understand it is not a simple query but a concept which depends on another concept. So the dependencies between the various concepts are not comprehensible.

**Comment by obfischer** [comment]
Hi @jdufner, I added some comments. If you like the can phone (mobile, skype) to speed up the review. ;-)

**Comment by DirkMahler** [comment]
Looks good, thanks @jdufner! Just one thing missing: the tutorial needs to be listed in the start page...

I'll add it "officially" in the beginning of the next week.

**Pull request merged**